### PR TITLE
Refactor central-application-connectivity-validator cache sync controller.

### PR DESCRIPTION
**Pull request by everesio**
**Description**

Refactor cache sync to use controller-runtime manager instead of cache lister

### CODE REVIEW

File: components/central-application-connectivity-validator/cmd/centralapplicationconnectivityvalidator/options.go [hunk 1]
```diff
@@ -117,5 +112,8 @@ func (o *options) validate() error {
 	if !strings.Contains(o.appRegistryPathPrefix, o.appNamePlaceholder) {
 		return fmt.Errorf("appRegistryPathPrefix '%s' should contain appNamePlaceholder '%s'", o.appRegistryPathPrefix, o.appNamePlaceholder)
 	}
+	if o.syncPeriod > time.Duration(o.cacheExpirationSeconds)*time.Second {
```
Comment (franpog859) [inline components/central-application-connectivity-validator/cmd/centralapplicationconnectivityvalidator/options.go:115]: I'd be super nice to add it also to the options' description in the `*.md` file and at the beginning of the `parseOptions()` function so an user knows about this validation from the start

**Review by franpog859 — COMMENTED** [review]
I like it a lot 🎊 Proceeding to the manual tests

**Review by franpog859 — COMMENTED** [review]
After registering the second Application I got this while trying to get its services:
```
$ curl https://gateway.somedomain.com/test-app/v1/metadata/services --cert generated.crt --key generated.key -k

{"code":500,"error":"while getting application ClientIds: application with name test-app is not found in the cache. Please retry"}
```

**Comment by everesio** [comment]
> After registering the second Application I got this while trying to get its services:
> 
> ```
> $ curl https://gateway.somedomain.com/test-app/v1/metadata/services --cert generated.crt --key generated.key -k
> 
> {"code":500,"error":"while getting application ClientIds: application with name test-app is not found in the cache. Please retry"}
> ```

Yes. this is exactly kind of problem what we had before refactoring. Seems like a problem with caching I've just added

```
ClientDisableCacheFor: []client.Object{
			&v1alpha1.Application{},
		},
		 
```

and I will check if it helps

File: components/central-application-connectivity-validator/cmd/centralapplicationconnectivityvalidator/options.go (hunk 1 above)
Comment (everesio) [reply components/central-application-connectivity-validator/cmd/centralapplicationconnectivityvalidator/options.go:115]: In the readme there is a section "Local cache refresh" which describes this restriction

**Comment by franpog859** [comment]
/test pre-main-kyma-gardener-azure-integration
See the previous logs [here](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-project_kyma/11846/pre-main-kyma-gardener-azure-integration/1423959097199300608). It was a failure regarding the Validator part

**Review by franpog859 — DISMISSED** [review]

**Review by majakurcius — APPROVED** [review]

**Review by majakurcius — APPROVED** [review]

**Review by franpog859 — APPROVED** [review]

**Pull request merged**